packages/std/src/math/decimal256.rs

```diff
@@ -0,0 +1,984 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::fmt::{self, Write};
+use std::ops;
+use std::str::FromStr;
+
+use crate::errors::StdError;
+
+use super::Fraction;
+use super::Isqrt;
+use super::Uint256;
+
+/// A fixed-point decimal value with 36 fractional digits, i.e. Decimal256(1_000_000_000_000_000_000) == 1.0
```

Comment:
Shouldn't this number be a constant (i.e. `MAX_PRECISION` or so) and be used across the impl?

Reply:
Probably! It crossed my mind, but I figured this PR is about making this work mostly consistently with `Decimal` rather than fixing the flaws that were already there.